python/src/iceberg/transforms.py

```diff
@@ -0,0 +1,520 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import base64
+import math
+import re
+import struct
+from datetime import datetime, timedelta
+from decimal import Decimal
+
+import mmh3  # type: ignore
+import pytz
+
+from .types import (
+    BinaryType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    Type,
+    UUIDType,
+)
+
+
+class Transform(object):
+    """
+    Transform base class for concrete transforms. The default implementation is for VoidTransform.
+    """
+
+    _EPOCH = datetime.utcfromtimestamp(0)
+
+    @staticmethod
+    def _human_day(day_ordinal):
```

Comment:
Do these need to be associated with the `Transform` class? Or should they be methods in the file that do not get exported?

Reply:
That is an internal method only used by subclasses of Transform so I did not promote them to the module method.
Or I may just inline it. which used in two places.

Reply:
I moved it to transform_util file as a module method.